### PR TITLE
feat(argo-rollouts): Added Deployment labels to values.yaml

### PR DESCRIPTION
**Pull request by mpeters0n**
My company enforces Deployment labels to track team ownership, cost allocations, etc.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by jmeridth** [comment]
@mpeters0n plesse sign your commits to adhere to DCO. Thank you for your contribution

File: charts/argo-rollouts/templates/controller/deployment.yaml [hunk 1]
```diff
@@ -9,7 +9,12 @@ metadata:
...
   name: {{ include "argo-rollouts.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
+  {{- with (mergeOverwrite (deepCopy .Values.global.deploymentLabels) .Values.controller.deploymentLabels) }}
   labels:
+    {{- range $key, $value := . }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
     app.kubernetes.io/component: {{ .Values.controller.component }}
     {{- include "argo-rollouts.labels" . | nindent 4 }}
```
Comment (yu-croco) [inline charts/argo-rollouts/templates/controller/deployment.yaml:19]: In order not to break existing labels, maybe like this? 🙋 

```yaml
  labels:
    {{- with (mergeOverwrite (deepCopy .Values.global.deploymentLabels) .Values.controller.deploymentLabels) }}
    {{- toYaml . | nindent 4 }}
    {{- end }}
    app.kubernetes.io/component: {{ .Values.controller.component }}
    {{- include "argo-rollouts.labels" . | nindent 4 }}
```

File: charts/argo-rollouts/templates/controller/deployment.yaml [hunk 2]
```diff
@@ -10,6 +10,9 @@ metadata:
   name: {{ include "argo-rollouts.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
   labels:
+  {{- range $key, $value := (mergeOverwrite (deepCopy .Values.global.deploymentLabels) .Values.controller.deploymentLabels) }}
+    {{ $key }}: {{ $value | quote }}
+  {{- end }}
```
Comment (mpeters0n) [inline charts/argo-rollouts/templates/controller/deployment.yaml:15]: I tested this with both .Values.global.deploymentLabels defined and undefined to verify backwards compatibility

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**